py_ecc/field_elements.py

```diff
@@ -64,33 +57,39 @@ def __radd__(self, other: IntOrFQ) -> "FQ":
 
     def __rsub__(self, other: IntOrFQ) -> "FQ":
         on = other.n if isinstance(other, FQ) else other
-        return FQ((on - self.n) % field_modulus)
+        return FQ((on - self.n) % self.field_modulus, self.curve_name)
 
     def __sub__(self, other: IntOrFQ) -> "FQ":
         on = other.n if isinstance(other, FQ) else other
-        return FQ((self.n - on) % field_modulus)
+        return FQ((self.n - on) % self.field_modulus, self.curve_name)
 
     def __div__(self, other: IntOrFQ) -> "FQ":
         on = other.n if isinstance(other, FQ) else other
         assert isinstance(on, int)
-        return FQ(self.n * inv(on, field_modulus) % field_modulus)
+        return FQ(
+            self.n * prime_field_inv(on, self.field_modulus) % self.field_modulus,
+            self.curve_name
+        )
 
     def __truediv__(self, other: IntOrFQ) -> "FQ":
         return self.__div__(other)
 
     def __rdiv__(self, other: IntOrFQ) -> "FQ":
         on = other.n if isinstance(other, FQ) else other
         assert isinstance(on, int), on
```

Comment:
still some leftover runtime assertions that should be converted to explicit exception raising.

Reply:
@pipermerriam most of these asserts here are only minimal, and should infact be replaced by `raise` exceptions as part of other functions like `__add__`, `__mul__` etc. Or we ignore them completely all together. I would say that we add those `Exceptions`, as it is an important check. But how about we keep that for another PR or issue? (Which I would be taking up) This is because once we get the basic structure clean, rest all can be done easily.

Reply:
@pipermerriam done!